src/java/org/streamingpool/core/service/streamid/MergedErrorStreamId.java

```diff
@@ -0,0 +1,46 @@
+package org.streamingpool.core.service.streamid;
+
+import org.streamingpool.core.service.StreamId;
+
+import java.util.Objects;
+
+/**
+ * Given a source {@link StreamId} it published all the (deflected) errors of its ancestors
+ */
+public class MergedErrorStreamId implements StreamId<Throwable> {
+    private static final long serialVersionUID = 1L;
+
+    private final StreamId<?> source;
+
+    private MergedErrorStreamId(StreamId<?> source) {
+        this.source = source;
+    }
+
+    public static MergedErrorStreamId mergeErrorsStartingFrom(StreamId<?> source) {
```

Comment:
how about `mergeErrorsUpstreamOf(...)`?

Reply:
@ciastkoMalinowe and anita, what do you think?